### DISCUSSION TITLE
Invalid scopes error

### DISCUSSION THREAD

**Discussion by Zyles** [keycloak/keycloak#34424] [Q&A]
I am a bit confused.

I setup my client with resources, scopes, roles, permissions.

Resources:
![image](https://github.com/user-attachments/assets/1743f14b-3791-4306-b4bc-3291904e7d9b)

Scopes:
![image](https://github.com/user-attachments/assets/ad49b16c-2f46-4b67-81d4-4ef8f40a1638)

Policies:
![image](https://github.com/user-attachments/assets/b01a7e35-8de7-40f8-8f0e-8c3ccefdbb2e)

Permissions:
![image](https://github.com/user-attachments/assets/e2580f1c-7a72-43e9-9ed7-93a7501f4aaa)

Evaluate:
![image](https://github.com/user-attachments/assets/4c58bedf-af7b-4597-851a-8241fee9e7d6)

When I run evaluate everything is permitted/denied as expected.

But when I hit the endpoint: `{{keycloak_url}}/realms/{{realm}}/protocol/openid-connect/token`

With `scope=routes:create`

I get hit with:
```
{
    "error": "invalid_scope",
    "error_description": "Invalid scopes: routes:create"
}
```

**Comment by alexejibishvili** [comment]
Same problem here!
Maybe we are doing something wrong, but c'mon, we have seen a lot and it's very painful to understand Keycloak and its setup.
I encounter more and more problems as I go further. Too bad!
